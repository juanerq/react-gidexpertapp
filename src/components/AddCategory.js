import { useState } from "react"
import PropTypes from 'prop-types'

const AddCategory = ({setCategory}) => {

  const [inputValue, setInputValue] = useState('')

  const handleInputValue = (e) => {
    setInputValue( e.target.value )
  }

  const handleSubmit = (e) => {
    e.preventDefault()

    if (inputValue.trim().length > 2) {
      setCategory( cats => [inputValue, ...cats])
      setInputValue('')
    }
  }

  return (
    <form onSubmit={ handleSubmit }>
      <p>{ inputValue }</p>
      <input 
        type='text'
        value={ inputValue }
        onChange={ handleInputValue }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategory: PropTypes.func.isRequired
}

export default AddCategory