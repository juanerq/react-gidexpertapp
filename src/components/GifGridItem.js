import PropTypes from 'prop-types'

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeInUp">
      <div className="container-img">
        <img src={ url } alt={ title }/>
      </div>
      <p>{ title }</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
