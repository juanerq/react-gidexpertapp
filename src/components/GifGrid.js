import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from "../hooks/useFetchGifs"
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

  const { data:images, loading } = useFetchGifs( category )

  return (
    <>
      <h3 className='animate__animated animate__fadeInDown'>{ category }</h3>
      { loading && <p className='animate__animated animate__fadeInDown'>Loading</p> } 
      <div className='container-cards'>
          {
            images.map(img => (
              <GifGridItem key={img.id} {...img} />
            ))
          }
      </div>  
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}