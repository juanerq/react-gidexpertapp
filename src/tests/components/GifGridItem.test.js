import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'

describe('Component <GifGridItem/>', () => {
  const title = 'Nezuko'
  const url = 'http.//localhost:10101/nezuko.jpg'

  const wrapper = shallow( <GifGridItem title={ title } url={ url } /> )

  test('It should display Gif Grid Item correctly', () => {

    expect( wrapper ).toMatchSnapshot()

  })

  test('It should have the title Nezuko', () => {

    const p = wrapper.find('p')

    expect( p.text().trim() ).toBe( title )
  })

  test('It should have the image with url and alt', () => {

    const img = wrapper.find('img')

    expect( img.prop('src') ).toBe( url )
    expect( img.prop('alt') ).toBe( title )
  })

  test('It Should have the class animate__fadeInUp', () => {

    const div = wrapper.find('.animate__fadeInUp')
    const className = div.prop('className')
    expect( className.includes('animate__fadeInUp') ).toBe( true )

  })

})