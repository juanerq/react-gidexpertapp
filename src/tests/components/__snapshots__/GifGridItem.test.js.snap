// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <GifGridItem/> It should display Gif Grid Item correctly 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <div
    className="container-img"
  >
    <img
      alt="Nezuko"
      src="http.//localhost:10101/nezuko.jpg"
    />
  </div>
  <p>
    Nezuko
  </p>
</div>
`;
